src/screens/Blik/index.js

```diff
@@ -15,11 +20,65 @@ function BlikScreen({ navigation }) {
       ),
     });
   }, [navigation]);
+
+  useEffect(() => {
+    if (countdown > 0) {
+      const timer = setInterval(() => {
+        setCountdown((prev) => prev - 1);
+      }, 1000);
+      return () => clearInterval(timer);
+    } else {
+      setIsCounting(!isCounting);
+    }
+  }, [countdown]);
+
+  function generateBlikNumber() {
+    return Array.from({ length: 6 }, () => Math.floor(Math.random() * 10)).join(
+      '',
+    );
+  }
+
+  const resetTimer = () => {
+    setIsCounting(!isCounting);
+    setBlikNumber(generateBlikNumber());
+    setCountdown(4);
+  };
+
   return (
-    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
-      <Text>BlikScreen</Text>
+    <View style={styles.container}>
+      {!isCounting ? (
+        <View style={styles.container}>
+          <Text style={{ fontSize: 24 }}>Code expired</Text>
+          <Button mode="outlined" onPress={resetTimer}>
+            New code
+          </Button>
+        </View>
+      ) : (
+        <View style={styles.container}>
+          <Text style={{ fontSize: 22 }}>Your code is:</Text>
+          <Text style={{ fontSize: 30, fontWeight: 'bold' }}>{blikNumber}</Text>
+          <Text style={{ fontSize: 20 }}>{countdown}s</Text>
```

Comment:
fajne, można dać jeszcze progress bar zamiast licznika
```suggestion
// max = 4 w tym przypadku
<ProgressBar progress={1 - (max - countdown) / max } style={{ width: '100%' }} />
```

Reply:
albo to i to w sumie